### DISCUSSION TITLE
Adonis [4.1] How use validator inside controller?

### DISCUSSION THREAD

**Discussion by RafaMoralles** [adonisjs/core#1645] [Help]
i have request like this

127.0.0.1:3333/companies

``` 
{
  "name": "comapnies",
  "photo": "code.png",
  "about": "Agency",
  "whatsapp": "51997156841",
  "timedelivery": 45,
  "address":
  {
	"cep": "00000000",
	"street": "street xx",
	"number": "220",
	"complement": "",
	"district": "Center",
	"city": "City",
	"state": "City"
  }
} 
```

And i have one validator like this

``` 'use strict'

class Address {
  async fails (error) {
    return this.ctx.response.status(400).send({ error: { message: error[0].message, name: 'PropertyRequired', status: 400 } })
  }

  get rules () {
    return {
      street: 'required',
      number: 'required',
      district: 'required',
      city: 'required'
    }
  }

  get messages () {
    return {
      'street.required': 'Informe a propriedade Rua',
      'number.required': 'Informe a propriedade Número',
      'district.required': 'Informe a propriedade Bairro',
      'city.required': 'Informe propriedade Cidade'
    }
  }
}

module.exports = Address
 ```

how can i use this validator Address in another controller?

**Comment by viniciuspsw** [comment]
You can use the validator in multiple routes by defining in each route definition, for example:

```js
Route.post('addresses', 'AddressController.store').validator('Address')
Route.post('second-controller', 'SecondController.store').validator('Address')
Route.post('third-controller', 'ThirdController.store').validator('Address')
```

Take a look at [Route validator docs](https://adonisjs.com/docs/4.1/validator#_route_validator).

**Reply by RafaMoralles** [reply]
Hi @viniciuspsw thanks for your reply but in my routes i use validator for companies, something like

Route.resource('companies', 'CompanyController').apiOnly()
  .validator(new Map([[['companies.store'], ['Company']], [['companies.update'], ['Company']]]))

but i need a second validator inside my controller, because my json have an atributte call address and i want to validate those informations. I have another route /address, with validator same validator i posted before.

i Try import validator like
```
const { ValidatorAddress } = use('App/Validators/Address')
const { validate } = use('Validator')

```

but doesn't work, any help will be great
